source/MaterialXGenOsl/OslSyntax.cpp

```diff
@@ -479,12 +479,11 @@ OslSyntax::OslSyntax()
     registerTypeSyntax
     (
         Type::DISPLACEMENTSHADER,
-        std::make_shared<OslStructTypeSyntax>(
+        std::make_shared<ScalarTypeSyntax>(
             "displacementshader",
-            "{vector(0.0), 0.0}",
-            "{vector(0.0), 0.0}",
-            EMPTY_STRING,
-            "struct displacementshader { vector offset; float scale; };")
+            "vector(0.0)",
+            "vector(0.0)",
+            "vector")
```

Comment:
Changed displacementshader type to be a single vector in OSL (displacement result). The displacementshader constructor node handles taking a displacement vector + a scalar weight, but the result is just a vector.